coffee-module/coffee-module-mp/coffee-module-mp-restclient/src/main/java/hu/icellmobilsoft/coffee/module/mp/restclient/exception/FaultTypeParser.java

```diff
@@ -46,15 +50,25 @@ public FaultTypeParser() {
      *            {@link String} to parse.
      * @return {@link hu.icellmobilsoft.coffee.dto.error.IFaultType} parsed from input string.
      */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
     public static Enum<?> parseFaultType(String faultTypeString) {
-        for (Class<? extends Enum> faultTypeClass : FaultTypeParserExtension.getFaultTypeClasses()) {
-            Enum<?> fault = EnumUtils.getEnum(faultTypeClass, faultTypeString);
-            if (fault != null) {
-                return fault;
+        Instance<IFaultTypeProvider> faultProviderInstance = CDI.current().select(IFaultTypeProvider.class);
```

Comment:
Az eredeti megoldas az interface vagy annotacio letezese szerint mukodott, most viszont kezzel fog kelleni listat epitani amibol rendszeresen ki lessz felejtve az uj enum class. Valahogy ezt a lista epiteset automatara kellene kesziteni

Reply:
igen, konzultaltam Markkal ezzel kapcsolatban, de nem nagyon sikerult olyat kitalalnil ami automata lenne.
(ugye nem kerul bele a bean discovery-be az enum igy max reflectionnel tudnank vegigmenni az osszes osztalyon, de ezt a modszert lattuk jobbnak)